CONTRIBUTING.md

```diff
@@ -0,0 +1,55 @@
+6. If the Pull Request is merged, it will create an official ACCESS-NRI released build of the given model, with an associated GitHub Release.
+
+## Modifications To The `spack.yaml` File
+
+Below are some real-world examples of how to modify the `spack.yaml` file.
+
+### Example Modification Workflow of `ACCESS-OM2`s `MOM5` Package
+
+In this example, we are using the `ACCESS-OM2` deployment repositories `spack.yaml` from [this commit](https://github.com/ACCESS-NRI/ACCESS-OM2/blob/47bc7bf979c1dfa12a24272cb739117abc50d7ca/spack.yaml). We are also modifying one of ACCESS-OM2's model components, namely [ACCESS-NRI/MOM5](https://github.com/ACCESS-NRI/MOM5). This model component is available in our [`ACCESS-NRI/spack-packages`](https://github.com/ACCESS-NRI/spack-packages) as the package `mom5`.
+
+1. Clone and make modifications to the model component `ACCESS-NRI/MOM5`, which is tagged with [`dev_2024.08.14`](https://github.com/ACCESS-NRI/MOM5/releases/tag/dev_2024.08.14).
+2. Create a branch in the model deployment repository `ACCESS-NRI/ACCESS-OM2` - such as `update_mom5_dev_build`.
+3. Make modifications to the `spack.yaml` file:
+   * Firstly, bump the version of the [overall deployment](https://github.com/ACCESS-NRI/ACCESS-OM2/blob/47bc7bf979c1dfa12a24272cb739117abc50d7ca/spack.yaml#L8). This version encompasses all changes done in a `spack.yaml`, in the form `CALVER_YEAR.CALVER_MONTH.MINOR`:
+      * Since this is a minor change, just bump the minor version. Hence, it is updated to `@git.2024.03.1`.
+      * When that version is bumped, the [associated module projection](https://github.com/ACCESS-NRI/ACCESS-OM2/blob/47bc7bf979c1dfa12a24272cb739117abc50d7ca/spack.yaml#L51) also needs to be updated to `{name}/2024.03.1` as well.
+   * Secondly, bump the version of `mom5` that was updated earlier, to the `dev_2024.08.14` tag set in the model component repository:
+      * Update the version of the [`mom5` package](https://github.com/ACCESS-NRI/ACCESS-OM2/blob/47bc7bf979c1dfa12a24272cb739117abc50d7ca/spack.yaml#L15) to `@git.dev_2024.08.14`.
+      * Also update the [associated module projection](https://github.com/ACCESS-NRI/ACCESS-OM2/blob/47bc7bf979c1dfa12a24272cb739117abc50d7ca/spack.yaml#L53) to `{name}/VERSION-{hash:7}` - in this case, `{name}/dev_2024.08.14-{hash:7}`. The `{hash:7}` is used so the module doesn't conflict with other versions.
+4. Now that the modifications are complete, commit the changes and open a Pull Request in `ACCESS-NRI/ACCESS-OM2` into `main`, even if it's just a dev build.
+5. The CI/CD kicks off, and a comment is added by `github-actions[bot]` saying that the build is deploying, and once the deployment is 'Active', the module specified in the comment is available on the HPC and is `module use/load`able. Excellent!
```

Comment:
Do you want to show an example of a comment that was generated? Is it worth having a Point 6 on logging into Gadi and running the commands?

Reply:
> Do you want to show an example of a comment that was generated? 

Yep, will add an example. 

> Is it worth having a Point 6 on logging into Gadi and running the commands?

I suppose so, but what happens in future when we have multiple deployment targets? I think I'd rather keep the instructions general to some HPC rather than a specific one

Reply:
Done the example in https://github.com/ACCESS-NRI/model-deployment-template/pull/9/commits/3b85309a19b45bcd7779340bea3011388504ad9e, will wait for feedback on the other point.

Reply:
> Is it worth having a Point 6 on logging into Gadi and running the commands?

Feels like a stretch for this document, and would add more delays to getting it merged. I think it is something a hive docs version could address later.